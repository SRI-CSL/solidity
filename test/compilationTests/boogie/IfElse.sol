pragma solidity ^0.4.17;

contract IfElse {
    uint contractVar;

    function doSomething(uint param) view public returns (uint) {
        if (param < 10) {
            return contractVar + 10;
        } else if (param < 20) {
            return contractVar + 20;
        } else {
            return contractVar;
        }
    }

    function doSomethingElse(uint param) view public returns (uint) {
        if (param < 10) {
            return contractVar + 10;
        }

        return contractVar;
    }

    function conditional(uint param) pure public returns (uint) {
        return param > 100 ? 100 : param;
    }
}