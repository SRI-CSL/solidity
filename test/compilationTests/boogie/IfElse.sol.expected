Verifying IfElse::__verifier_main: OK
No errors found.
